### ISSUE TITLE
Feature request:

### ISSUE THREAD

**Issue by Tzvook** [lokothodida/gs-catalog#10]
I'm not sure  if this is the right place ( is it the "pull requests" ? ) , but I thought it'll be a good idea to post somewhere a list of ideas for improvement:
I simply implemented the catalog in a real site I'm making, it's the best way to "feel the real need"  
- main category screen - hierarchical view.
- Ability to define custom/edit category slugs
- "new category" button from the "edit category" screen too.
- Multi-lang support ???

**Comment by lokothodida** [comment]
Yep, this is the place ([pull requests](https://github.com/lokothodida/gs-catalog/pulls) is for merging branches).
1. Can do.
2. Can do.
3. Can do.
4. Guessing you mean for the front end where the user can select the language, and the admin can define langauge-specific content for fields (like with i18n)? Would you want the administrator to have their own catalog hashes for words? E.g. a page called i18n where they will fill in hash fields and access them in templates going `catalog_i18n('YOUR_WORD')`

**Comment by Tzvook** [comment]
Yep - the "i18n way" (by the way - the automatic browser-lang selection works well with the i18n- I tested it with 4 languages under chrome+IE+FF+Opera, so in most cases the user don't need to choose his language)
About the back-end ... it can be in the lang file, but . . . it seems that this catalog will give GS new standards anyway :-)

by the way - main category screen - hierarchical view - I meant in Admin ...

**Comment by lokothodida** [comment]
Okay. I'll need to look at mvlcek's source code to see how he's integrated it to be sure it works in the same way.

I don't know about giving GS new standards, haha. We'll need to see how this goes.

Understood.

I'm currently doing a major refactoring/rewrite of the files in the /lib directory which I haven't yet pushed the changes for. If I think it goes somewhere positive, they'll be pushed on a new branch, worked on more and then merged into v0.3. If not, I'll continue with v0.3 as it currently is.

**Comment by Tzvook** [comment]
Is there something testable ?

**Comment by lokothodida** [comment]
Not just yet. I've refactored a lot of the code, and am in the process of integrating i18n for the products. I should have something some time next week.

**Comment by Tzvook** [comment]
Can I use the new branch ? sorry about the nagging :-)

**Comment by lokothodida** [comment]
Of the 4 requests, 2 are fully implemented, 1 is in progress (multi-language), and the slug renaming is the one I am hesitant to implement because it involves ensuring that all multi-language pages are also renamed and clashes are avoided.

What would the expectations be of the renaming function?
